examples/program.c

```diff
@@ -36,6 +44,37 @@ int main(int argc, char *argv[])
         jl_value_t *s = (jl_value_t*)jl_cstr_to_string(argv[i]);
         jl_arrayset(ARGS, s, i - 1);
     }
+
+    //printf("here\n");
+    //jl_binding_t *bp = jl_get_binding_wr(jl_base_module, jl_symbol("nathan"));
+    //bp->value = (jl_value_t*)jl_cstr_to_string("I <3 NATHAN");
+    //jl_gc_wb(jl_base_module, bp->value);
+
+
+    //jl_value_t *n = (jl_value_t*)jl_get_global(jl_base_module, jl_symbol("nathan"));
+    //printf("nathan: %s\n", jl_string_ptr(n));
+
+
+    // Pass command line args to julia.
+    jl_set_ARGS(argc, argv);  // Sets Core.ARGS to argv.
+    // Set PROGRAM_FILE to argv[0], to match behavior of Base._start().
+    jl_set_global(jl_base_module,
+        jl_symbol("PROGRAM_FILE"),
+        (jl_value_t*)jl_cstr_to_string(argv[0]));
+    // Set Base.ARGS to argv[1:end], to match behavior of Base._start().
+    //  -- jl_set_const doesn't work because of https://github.com/JuliaLang/julia/pull/27520.
+    //jl_set_const(jl_base_module, jl_symbol("ARGS"), (jl_value_t*)ARGS);
+    //  -- This should work, but for some reason it doesn't. It says it's
+    //  -- "WARNING: redefining constant ARGS", but for w/e reason doesn't...
+    //  -- I don't understand why it's not working.
+    //jl_checked_assignment(jl_get_binding_wr(jl_base_module, jl_symbol("ARGS")),
+    //                       (jl_value_t*)ARGS);
+    // -- Trying this instead, even though it's bulky:
+    set_julia_base_ARGS(argc, argv);
```

Comment:
So all of the above is really wonky... I first just tried overwriting the Base.ARGS via `jl_set_const`, but ran into this problem:
https://github.com/JuliaLang/julia/pull/27520

So then I tried overwriting it via `jl_checked_assignment`, which really should work, but for some reason the compiled program still has an empty `String[]` for ARGS, and i don't know why...

So finally, i've caved and i'm just manually appending all the fields to the Base.ARGS, just like `Base._start()` does, via this helper function `set_julia_base_ARGS`.